### ISSUE TITLE
Feature request: add responsive btn-group-vertical classes

### ISSUE THREAD

**Issue by Rabadash8820** [twbs/bootstrap#39213]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Add responsive `btn-group-*` and/or `btn-group-*-vertical` classes, that cause a `btn-group` to appear vertical or horizontal at the specified breakpoints (`sm`, `md`, etc.).

### Motivation and context

`btn-group` is a handy component for visually/connecting multiple buttons. However, if there are lots of buttons with text (especially when using `btn-group-lg`), then the buttons will be awkwardly squished together.

For example, here is a 3-button group on a widescreen laptop:

![image](https://github.com/twbs/bootstrap/assets/8388785/54e0db71-0e13-4f27-847e-7ed50af5d4c4)

and again on a mobile screen:

![image](https://github.com/twbs/bootstrap/assets/8388785/e8e7b371-d638-4432-a0f0-3587c45261dc)

[`btn-group-vertical`](https://getbootstrap.com/docs/5.3/components/button-group/#vertical-variation) looks much nicer on mobile, but leads to a lot of unnecessary surrounding whitespace on widescreen:

![image](https://github.com/twbs/bootstrap/assets/8388785/7ac7d38d-abd2-4f99-850f-a8bbedf876b4)

This [SO answer](https://stackoverflow.com/questions/49176385/make-btn-group-responsive) with over 70 upvotes suggests using `btn-group flex-wrap`. However, this makes the button corners look weird on mobile without several additional `border-radius-*` utilities:

![image](https://github.com/twbs/bootstrap/assets/8388785/dddacd2e-232e-4201-b521-6417312dfbdb)

I could also use [block buttons](https://getbootstrap.com/docs/5.3/components/buttons/#block-buttons), but they lack the nice connected borders of a button group.

Long story short, if I could just put `btn-group-vertical btn-group-md` on my button groups so that they start vertical on mobile and then expand to horizontal on `md` breakpoints and above, it would make everything so much easier.

**Edit**: I know the proposed classes would conflict with the existing `btn-group-lg` and `btn-group-sm` classes for sizing... I don't have a great solution to suggest for that, other than making a breaking change or introducing more verbose classes like `btn-group-sm-horizontal`.

**Comment by Rabadash8820** [comment]
@mdo I realize that this Issue is old and relatively unnoticed by the community, but I'd like to argue that it is still an Issue that _should_ be fixed eventually. The current behavior is rather poor for responsive button groups with many buttons. I request re-opening this one . 🙏

**Comment by mdo** [comment]
Woof, this would be a doozy to do with a ton of extra CSS. Maybe we could make it something you opt into with a custom Sass build.

**Comment by Rabadash8820** [comment]
I'm surprised to hear that this would require so much extra CSS, but yeah, any solution/workaround that you can provide would be much appreciated. 🙏

**Comment by mdo** [comment]
It's straightforward to do, but because we generate styles for each breakpoint, that code repeats 5-6 times over :). That's all I meant.

**Comment by Rabadash8820** [comment]
Ah I see. Speaking as one who uses BS as-is without custom Sass builds, having these classes available out of the box would be ideal, but I respect the impulse to keep things DRY
